jujugui/static/gui/src/app/components/entity-files/entity-files.js

```diff
@@ -0,0 +1,110 @@
+/*
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
+Copyright (C) 2015 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+YUI.add('entity-files', function() {
+
+  juju.components.EntityFiles = React.createClass({
+    /* Define and validate the properites available on this component. */
+    propTypes: {
+      entityModel: React.PropTypes.object.isRequired,
+      pluralize: React.PropTypes.func.isRequired
+    },
+
+    /**
+      If able, generates a link to the entity's source code.
+
+      @method _generateCodeLink
+      @param {Object} codeSource metadata about the entity's source code.
+      @return {Object} The markup for the link to the code.
+    */
+    _generateCodeLink: function(codeSource) {
```

Comment:
The point here is to make sure that `codeSource.location` never fails because codeSource is `null`; even if `null` were explicitly passed in, we need to override it to `{}`.